sdk/javascript/test/BaseValue_spec.js

```diff
@@ -0,0 +1,123 @@
+const { BaseValue } = require('../src/BaseValue');
+const converter = require('../src/utils/converter');
+const { expect } = require('chai');
+
+describe.only('BaseValue', () => {
+	const DEFAULT_VALUE = BigInt('0x456789ABCDEF0123');
+
+	class MockValue extends BaseValue {
+		constructor(value = 0) {
+			super(MockValue.SIZE, value);
+		}
+
+		static deserialize(arrayInput) {
+			return new MockValue(converter.intFromBytes(arrayInput, MockValue.SIZE, false));
+		}
+	}
+	MockValue.SIZE = 8;
+
+	const canCreateBaseValueFactory = isSigned => (value, size, expectedValue) => {
+		// Act:
+		const val = new BaseValue(size, value, isSigned);
+
+		// Assert:
+		expect(val.size()).to.equal(size);
+		expect(val.value).to.equal(expectedValue);
+	};
+
+	const canCreateUnsignedBaseValue = canCreateBaseValueFactory(false);
+	const canCreateSignedBaseValue = canCreateBaseValueFactory(true);
+
+	it('cannot create 64-bit value with invalid type', () => {
+		// Assert:
+		expect(() => new BaseValue(8, -1)).to.throw('has invalid type');
+		expect(() => new BaseValue(8, 1)).to.throw('has invalid type');
+	});
+
+	it('cannot create from dumb values', () => {
+		// Assert:
+		[1, 2, 4].forEach(byteSize => {
+			expect(() => new BaseValue(byteSize, NaN)).to.throw('is not an integer');
+			expect(() => new BaseValue(byteSize, -Infinity)).to.throw('is not an integer');
+			expect(() => new BaseValue(byteSize, Infinity)).to.throw('is not an integer');
+			expect(() => new BaseValue(byteSize, undefined)).to.throw('is not an integer');
+		});
+	});
+
+	it('cannot create unsigned with values outside range', () => {
+		// Assert:
+		expect(() => new BaseValue(1, -1)).to.throw('outside of valid 8-bit range');
```

Comment:
can you add 3rd test of some value in between for each (could also loop over values)

Reply:
not sure if I get you right, but don't you mean next test? `can create unsigned base value`

Reply:
i meant, something like this 
```
		canCreateSignedBaseValue(-0x80, 1, -0x80);
		canCreateSignedBaseValue(0x21, 1, -0x80);
		canCreateSignedBaseValue(0x7F, 1, 0x7F);
```

Reply:
oh, the comment was next to _cannot_, and  I couldn't figure out what you mean 😁